File: wizards/com/sun/star/wizards/web/data/CGContent.java
/**************************************************************
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 *
 *************************************************************/


package com.sun.star.wizards.web.data;

import com.sun.star.wizards.common.*;

import org.w3c.dom.*;

public class CGContent extends ConfigSetItem implements XMLProvider
{

    public String dirName;
    public String cp_Name;
    public String cp_Description;
    public ConfigSet cp_Contents = new ConfigSet(CGContent.class);
    public ConfigSet cp_Documents = new ConfigSet(CGDocument.class);

    public CGContent()
    {
        /*cp_Documents = new ConfigSet(CGDocument.class) {
        protected DefaultListModel createChildrenList() {
        return cp_Contents.getChildrenList();
        }
        };*/
    }

    public Node createDOM(Node parent)
    {

        Node myElement = XMLHelper.addElement(parent, "content",
                new String[]
                {
                    "name", "directory-name", "description", "directory"
                },
                new String[]
                {
                    cp_Name, dirName, cp_Description, dirName
                });

        cp_Documents.createDOM(myElement);

        return myElement;
    }
}

File: wizards/com/sun/star/wizards/web/data/CGContent.py
#
# This file is part of the LibreOffice project.
#
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.
#
# This file incorporates work covered by the following license notice:
#
#   Licensed to the Apache Software Foundation (ASF) under one or more
#   contributor license agreements. See the NOTICE file distributed
#   with this work for additional information regarding copyright
#   ownership. The ASF licenses this file to you under the Apache
#   License, Version 2.0 (the "License"); you may not use this file
#   except in compliance with the License. You may obtain a copy of
#   the License at http://www.apache.org/licenses/LICENSE-2.0 .
#
from ...common.ConfigGroup import ConfigGroup
from ..WebConfigSet import WebConfigSet
from ...common.XMLHelper import XMLHelper
from .CGDocument import CGDocument

class CGContent(ConfigGroup):

    def __init__(self):
        self.cp_Index = -1
        self.dirName = str()
        self.cp_Name = str()
        self.cp_Description = str()
        self.cp_Documents = WebConfigSet(CGDocument)
        self.cp_Contents = WebConfigSet(CGContent)

    def createDOM(self, parent):
        myElement = XMLHelper.addElement(
            parent, "content",
            ["name", "directory-name", "description", "directory"],
            [self.cp_Name, self.dirName, self.cp_Description, self.dirName])
        self.cp_Documents.createDOM(myElement)
        return myElement
